src/Data/Text.hs

```diff
@@ -1176,6 +1179,19 @@ minimum :: HasCallStack => Text -> Char
 minimum t = S.minimum (stream t)
 {-# INLINE minimum #-}
 
+-- TODO
+isAscii :: Text -> Bool
+isAscii (Text (A.ByteArray arr) (Exts.I# off#) (Exts.I# len#)) =
+    cSizeToInt (c_is_ascii_2 arr off# len#) == (Exts.I# len#)
+{-# INLINE isAscii #-}
+
+cSizeToInt :: CSize -> Int
+cSizeToInt = P.fromIntegral
+{-# INLINE cSizeToInt #-}
+
+foreign import ccall unsafe "_hs_text_is_ascii_2" c_is_ascii_2
+    :: ByteArray# -> Exts.Int# -> Exts.Int# -> CSize
```

Comment:
you don't need to use `Int#` here. `CSize` works fine (and is actually more correct).

Reply:
OK, thanks